frontend/src/app/shared/components/file-links/file-link-list/file-link-list-item.component.ts

```diff
@@ -106,4 +111,24 @@ export class FileLinkListItemComponent implements OnInit, AfterViewInit {
       );
     }
   }
+
+  private derivePermissionState():void {
+    switch (this.fileLink._links.permission.href) {
+      case 'urn:openproject-org:api:v3:file-links:permission:View':
+        this.showFloatingActions = true;
+        this.text.floatingText = '';
+        break;
+      case 'urn:openproject-org:api:v3:file-links:permission:NotAllowed':
```

Comment:
We have a capabilities API with which you can fetch permissions for a user. Is there a reason you did not use that one here?

Reply:
Those are permissions fetched from nextcloud. we cannot deliver those via the capabilities API.

Reply:
That's too bad :/ In that case I would either extract them to global constants so that you only have to manage one place. Or you create a service that does handles these special permissions.

Reply:
extracted to constants